projects/canopy/src/lib/filter-buttons/filter-button/filter-button.component.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  Component,
+  HostBinding,
+  Input,
+  OnInit,
+  Optional,
+  Self,
+  ViewEncapsulation,
+  Host,
+  SkipSelf,
+  ContentChild,
+} from '@angular/core';
+import { NgControl, ControlValueAccessor, FormGroupDirective } from '@angular/forms';
+
+import { LgDomService } from '../../utils/dom.service';
+import { LgErrorStateMatcher } from '../../forms/validation/error-state-matcher';
+import { LgValidationComponent } from '../../forms/validation/validation.component';
+import { LgFilterGroupComponent } from '../filter-group.component';
+import { FilterVariant } from '../filter-button.interface';
+
+let nextUniqueId = 0;
+
+@Component({
+  selector: 'lg-filter-button',
+  templateUrl: './filter-button.component.html',
+  styleUrls: ['./filter-button.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class LgFilterButtonComponent implements OnInit, ControlValueAccessor {
+  @Input() checked = false;
+  @Input() id = `lg-filter-button-${++nextUniqueId}`;
```

Comment:
The `id` and `name` don't match because you are increasing the `nextUniqueId` in both places. You might want to save the increased value into a variable and then refer to that.

Reply:
Nice one. Done.